core/vm/contracts.go

```diff
@@ -145,13 +153,20 @@ func init() {
 	for k := range PrecompiledContractsCancun {
 		PrecompiledAddressesCancun = append(PrecompiledAddressesCancun, k)
 	}
+	for k := range PrecompiledContractsP256Verify {
+		PrecompiledAddressesP256Verify = append(PrecompiledAddressesP256Verify, k)
+	}
 }
 
 // ActivePrecompiles returns the precompiles enabled with the current configuration.
 func ActivePrecompiles(rules params.Rules) []common.Address {
 	switch {
 	case rules.IsArbitrum:
-		return PrecompiledAddressesArbitrum
+		var ret []common.Address
+		if rules.ArbOSVersion >= 30 {
+			ret = append(ret, PrecompiledAddressesP256Verify...)
```

Comment:
- This is called every tx, so I think we should consider caching this as PrecompiledAddressesArbOS30 or such
- We also need to update the EVM's `precompile` method in core/vm/evm.go

Reply:
Good catch, updated evm's precompile as well and introduced PrecompiledAddresses[/Contracts]ArbOS30 that is populated in geth hook (https://github.com/OffchainLabs/nitro/pull/2239).